File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int s = 0, e = people.length-1;
        Arrays.sort(people);
        int answer = 0;
        
        while(s<=e) {
            int temp = people[s]+people[e];
            if(temp <= limit) {
                s++;
                e--;
            } else {
                e--;
            }
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> people, int limit) {
    sort(people.begin(), people.end());
    
    int answer = 0;
    int s = 0, e = people.size()-1;
    while(s <= e) {
        int temp = people[s] + people[e];
        answer++;
        if(temp <= limit) {
            s++;
            e--;
        } else {
            e--;
        }
    }
    return answer;
}